include/boost/geometry/strategies/geographic/area_geographic.hpp

```diff
@@ -0,0 +1,212 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2016 Oracle and/or its affiliates.
+// Contributed and/or modified by Vissarion Fisikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_STRATEGIES_GEOGRAPHIC_AREA_GEOGRAPHIC_HPP
+#define BOOST_GEOMETRY_STRATEGIES_GEOGRAPHIC_AREA_GEOGRAPHIC_HPP
+
+#include <boost/geometry/formulas/area_formulas.hpp>
+#include <boost/geometry/formulas/thomas_inverse.hpp>
+//#include <boost/math/special_functions/atanh.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace area
+{
+
+/*!
+\brief Geographic area calculation by trapezoidal rule plus integral
+approximation that gives the ellipsoidal correction
+
+}
+
+*/
+
+template
+<
+    typename PointOfSegment,
+    typename Strategy = void,
+    const std::size_t SeriesOrder = 2,
+    bool ExpandEpsN = true,
+    bool LongSegment = false,
+    typename Spheroid = void,
+    typename CalculationType = void
+>
+class area_geographic
+{
+
+    //Select default types for Strategy, Spheroid and CalculationType
+    //in case they are not set
+
+    typedef typename boost::mpl::if_c
+    <
+        boost::is_void<CalculationType>::type::value,
+        typename select_most_precise
+            <
+                typename coordinate_type<PointOfSegment>::type,
+                double
+            >::type,
+        CalculationType
+    >::type CT;
+
+    typedef typename boost::mpl::if_c
+    <
+        boost::is_void<Strategy>::type::value,
+        typename geometry::formula::thomas_inverse
+            <
+                CT,
+                false,
+                true,
+                true
+            >,
+        Strategy
+    >::type AzimuthStrategy;
+
+    typedef typename boost::mpl::if_c
+    <
+        boost::is_void<Spheroid>::type::value,
+        geometry::srs::spheroid<CT>,
+        Spheroid
+    >::type SpheroidType;
+
+protected :
+    struct spheroid_constants
+    {
+        SpheroidType m_spheroid;
+        CT const m_a2;  // squared equatorial radius
+        CT const m_e2;  // squared eccentricity
+        CT const m_ep2; // squared second eccentricity
+        CT const m_ep;  // second eccentricity
+        CT const m_c2;  // authalic radius
+
+        inline spheroid_constants(SpheroidType spheroid)
+            : m_spheroid(spheroid)
+            , m_a2(math::sqr(get_radius<0>(spheroid)))
+            , m_e2(detail::flattening<CT>(spheroid)
+                 * (CT(2.0) - CT(detail::flattening<CT>(spheroid))))
+            , m_ep2(m_e2 / (CT(1.0) - m_e2))
+            , m_ep(math::sqrt(m_ep2))
+            , m_c2((m_a2 / CT(2.0)) +
+              ((math::sqr(get_radius<2>(spheroid)) * atanh(math::sqrt(m_e2)))
```

Comment:
What is this problem?
Anyway, `bg::math::atanh()` could be written in a way allowing to use ttmath's `atanh()` if defined instead ot STD version in C++11 or Boost.Math version in C++98 by use of `using` directive to bring a function into the scope. See e.g. how `bg::sqrt()` is defined: https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/util/math.hpp#L262